base/inference.jl

```diff
@@ -614,12 +614,14 @@ add_tfunc(Core.sizeof, 1, 1,
               x !== DataType && isleaftype(x) && return _const_sizeof(x)
```

Comment:
> isType(x) && return _const_sizeof(x.parameters[1])

This case isn't valid unless we define that `T == S` implies `sizeof(T) === sizeof(S)`, which currently is not true. Either we need to use `isConstType` here (such that we know `T == S` implies `T === S`) or work out all of the equivalence classes that could occur (tempting, since we seem to be really close, but probably difficult to get right)